### PR TITLE
fix(ds-matching-list): fallback to previous implementation if needed

### LINKED ISSUE #2084: Zowe Explorer API's dataSetsMatchingPattern error found on Che
**Describe the bug**

An error was found when performing a filter search on a Che environment for the DS view.

**To Reproduce**

1. Create unsecure team config profile on a Che environment
2. Reload window
3. click on a profile and perform a filter search on the DS view
4. enter credentials when prompted
5. do any filter search e.g User.*
6. error should appear

**Expected behavior**

Should display items when performing a filter search successfully

**Screenshots**

Screen Shot 2023-01-13 at 1 18 58 PM

**Desktop (please complete the following information):**

- OS:
- Zowe Explorer Version:
- (Optional) Zowe CLI Version:
- (Optional) Are you using Secure Credential Store?

**Additional context**

Could be potentially related to #2038 
@zFernand0

### PR DESCRIPTION
**Pull request by zFernand0**
## Proposed changes

- Resolves #2084 

## Release Notes

Fallback to the previous implementation when there is no implementation for `dataSetsMatching` (`dsm`)

Milestone: 2.6.0

Changelog: Fallback to previous implementation when `dsm` is not implemented in the `mvsApi`

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [x] All checks have passed (DCO, Jenkins and Code Coverage)
- [x] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [x] There is coverage for the code that I have added
- [x] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [x] Any PR dependencies have been merged and published (if appropriate)

### CODE REVIEW

**Review by traeok — APPROVED** [review]

File: packages/zowe-explorer/src/dataset/ZoweDatasetNode.ts [hunk 1]
```diff
@@ -282,19 +282,22 @@ export class ZoweDatasetNode extends ZoweTreeNode implements IZoweDatasetTreeNod
...
+                            .split(",")
+                            .map((p) => p.trim())
+                    ),
+                ];
+                const mvsApi = ZoweExplorerApiRegister.getMvsApi(cachedProfile);
+                if (mvsApi.dataSetsMatchingPattern) {
+                    responses.push(await mvsApi.dataSetsMatchingPattern(dsPatterns));
+                } else {
+                    for (const dsp of dsPatterns) {
+                        responses.push(await mvsApi.dataSet(dsp));
```
Comment (traeok) [inline packages/zowe-explorer/src/dataset/ZoweDatasetNode.ts:298]: We might want to add patch coverage for this scenario to verify the correct function is called based on this condition, but your logic here should definitely handle that.

**Review by rudyflores — APPROVED** [review]
Tested this PR on Che and it seems to be all good now, thanks @zFernand0 !

File: packages/zowe-explorer/src/dataset/ZoweDatasetNode.ts (hunk 1 above)
Comment (zFernand0) [reply packages/zowe-explorer/src/dataset/ZoweDatasetNode.ts:299]: Added such test to make sure that the old API is still being called.
Sorry for the delay

**Review by traeok — APPROVED** [review]
LGTM, thanks @zFernand0 !

**Review by rudyflores — APPROVED** [review]

**Pull request merged**